### ISSUE TITLE
GHC 9.6.1 support

### ISSUE THREAD

**Issue by dhess** [ocharles/logging-effect#37]
Hi @ocharles, thanks for this library, it's just the sort of thing we needed for logging in our Haskell application.

Can we get some fixes for GHC 9.6.1? Or perhaps I should open an issue on https://github.com/nikita-volkov/list-t instead, if you prefer?

```
	Preprocessing library for logging-effect-1.3.13..
Building library for logging-effect-1.3.13..
 
src/Control/Monad/Log.hs:2:14: warning: [-Wdeprecated-flags]
    -XAutoDeriveTypeable is deprecated: Typeable instances are created automatically for all types since GHC 8.2.
  |
2 | {-# LANGUAGE AutoDeriveTypeable #-}
  |              ^^^^^^^^^^^^^^^^^^
installing
[1 of 1] Compiling Control.Monad.Log ( src/Control/Monad/Log.hs, dist/build/Control/Monad/Log.o, dist/build/Control/Monad/Log.dyn_o )
post-installation fixup
checking for references to /build/ in /nix/store/8ymg1pmv922fljjxwwj7bpcbjn26k4sh-list-t-lib-list-t-1.0.5.6...
Installing library in /nix/store/pn7823xwccg2fifhgd7c24gvb6c5khbs-deferred-folds-lib-deferred-folds-0.9.18.3/lib/x86_64-linux-ghc-9.6.1/deferred-folds-0.9.18.3-7LT9EhGPfOz6ZQdKpqFKND
patchelf: wrong ELF type
patching script interpreter paths in /nix/store/8ymg1pmv922fljjxwwj7bpcbjn26k4sh-list-t-lib-list-t-1.0.5.6
 
checking for references to /build/ in /nix/store/lh7xijx35xcnmrkznkg94i6pg5ffyd4r-list-t-lib-list-t-1.0.5.6-configFiles...
src/Control/Monad/Log.hs:121:1: error:
    Could not find module ‘Control.Monad.Trans.Error’
    Perhaps you meant
      Control.Monad.Trans.Free (from free-5.2)
      Control.Monad.Trans.Iter (from free-5.2)
      Control.Monad.Trans.Writer (from transformers-0.6.1.0)
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
    |
121 | import qualified Control.Monad.Trans.Error as Error
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
src/Control/Monad/Log.hs:125:1: error:
    Could not find module ‘Control.Monad.Trans.List’
    Perhaps you meant
      Control.Monad.Trans.Cont (from transformers-0.6.1.0)
      Control.Monad.Trans.RWS (from transformers-0.6.1.0)
      Control.Monad.Trans.Free (from free-5.2)
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
    |
125 | import qualified Control.Monad.Trans.List as List
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
patching script interpreter paths in /nix/store/lh7xijx35xcnmrkznkg94i6pg5ffyd4r-list-t-lib-list-t-1.0.5.6-configFiles
checking for references to /build/ in /nix/store/m1f4y4vfrv5rffdirb1p3n0h99jbfi84-list-t-lib-list-t-1.0.5.6-ghc...
error: builder for '/nix/store/m5xjvl64893g2jr7bdhf46gf7zmw0b40-logging-effect-lib-logging-effect-1.3.13.drv' failed with exit code 1;
       last 10 log lines:
       > src/Control/Monad/Log.hs:125:1: error:
       >     Could not find module ‘Control.Monad.Trans.List’
       >     Perhaps you meant
       >       Control.Monad.Trans.Cont (from transformers-0.6.1.0)
       >       Control.Monad.Trans.RWS (from transformers-0.6.1.0)
       >       Control.Monad.Trans.Free (from free-5.2)
       >     Use -v (or `:set -v` in ghci) to see a list of the files searched for.
       >     |
       > 125 | import qualified Control.Monad.Trans.List as List
       >     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
       For full logs, run 'nix log /nix/store/m5xjvl64893g2jr7bdhf46gf7zmw0b40-logging-effect-lib-logging-effect-1.3.13.drv'.
error: 1 dependencies of derivation '/nix/store/w8n67pyx7xvifwxjcp7chi6asi53q6zb-primer-lib-primer-0.7.2.0.drv' failed to build
🚨 Error: The command exited with status 1
user command error: exit status 1

```

**Comment by brprice** [comment]
@dhess I believe this should be fixed as of #39 (released as 1.4.0)

**Comment by dhess** [comment]
This has indeed been fixed. Thanks!
